src/containerapp/azext_containerapp/custom.py

```diff
@@ -4297,3 +4306,195 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
+    else:
+        img_info = subprocess.Popen(pack_exec_path + " inspect-image " + img["imageName"] + " --output json", shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        img_info_out, img_info_err = img_info.communicate()
+        if img_info_err.find(b"status code 401 Unauthorized") != -1 or img_info_err.find(b"unable to find image") != -1:
+            inspect_result = dict(remote_info=401, image_name=img["imageName"])
+        else:
+            inspect_result = json.loads(img_info_out)
+        inspect_result.update({"targetContainerName": img["targetContainerName"], "targetContainerAppName": img["targetContainerAppName"], "targetContainerAppEnvironmentName": img["targetContainerAppEnvironmentName"], "targetResourceGroup": img["targetResourceGroup"]})
+    info_list.append(inspect_result)
+
+
+def patch_interactive(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.warning("Please install or start Docker and try again.")
+        return
+    patchable_check_results = patch_list(cmd, resource_group_name, managed_env, show_all=show_all)
+    pack_exec_path = get_pack_exec_path()
+    if patchable_check_results is None:
+        return
+    patchable_check_results_json = json.dumps(patchable_check_results, indent=2)
+    without_unpatchable_results = []
+    without_unpatchable_results = [result for result in patchable_check_results if result["id"] is not None]
+    if without_unpatchable_results == [] and (patchable_check_results is None or show_all is False):
+        return
+    print(patchable_check_results_json)
+    if without_unpatchable_results == []:
+        return
+    user_input = input("Do you want to apply all the patches or specify by id? (y/n/id)\n")
+    patch_apply(cmd, patchable_check_results, user_input, pack_exec_path)
+
+
+def patch_apply(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.warning("Please install or start Docker and try again.")
+        return
+    patchable_check_results = patch_list(cmd, resource_group_name, managed_env, show_all=show_all)
+    pack_exec_path = get_pack_exec_path()
+    if patchable_check_results is None:
+        return
+    patchable_check_results_json = json.dumps(patchable_check_results, indent=2)
+    without_unpatchable_results = []
+    without_unpatchable_results = [result for result in patchable_check_results if result["id"] is not None]
+    if without_unpatchable_results == [] and (patchable_check_results is None or show_all is False):
+        return
+    print(patchable_check_results_json)
+    if without_unpatchable_results == []:
+        return
+    patch_apply(cmd, patchable_check_results, "y", pack_exec_path)
+
+
+def patch_apply(cmd, patch_check_list, method, pack_exec_path):
+    m = method.strip().lower()
```

Comment:
@snehapar9 @harryli0108 -- nit: can we update the name of this variable so it's clear what the value represents? It would sense to either keep the `method` variable name or be renamed to something like `input_method`

Reply:
Thanks! just found some other issues in that block. fixing them all now